src/main/java/com/google/sps/servlets/RecipePostServlet.java

```diff
@@ -0,0 +1,183 @@
+import com.google.appengine.api.datastore.KeyRange;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.search.Document;
+import com.google.appengine.api.search.Field;
+import com.google.appengine.api.search.Index;
+import com.google.appengine.api.search.IndexSpec;
+import com.google.appengine.api.search.Results;
+import com.google.appengine.api.search.ScoredDocument;
+import com.google.appengine.api.search.SearchException;
+import com.google.appengine.api.search.SearchServiceFactory;
+import com.google.appengine.api.search.StatusCode;
+import com.google.gson.Gson;
+import java.io.*;
+import java.io.IOException;
+import java.lang.*;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.commons.lang3.StringUtils;
+
+@WebServlet("/recipe_post")
+public class RecipePostServlet extends HttpServlet {
+
+  /**
+   * doPost creates a new recipe entity with the attributes inputted in the post request
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) {
+    String title = request.getParameter("title");
+    String imgURL = "TODO Add url here";
+    String description = request.getParameter("description");
+    int prepTime = getPrepTime(request);
+    String difficulty = request.getParameter("difficulty");
+    ArrayList<String> ingredients = getIngredient(request);
+    ArrayList<String> stepList = getSteps(request);
+
+    IndexSpec indexSpec = IndexSpec
+      .newBuilder()
+      .setName("recipesIndex")
+      .build();
+    Index index = SearchServiceFactory.getSearchService().getIndex(indexSpec);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    KeyRange keyRange = datastore.allocateIds("Recipe", 1L);
+
+    Entity recipeEntity = buildRecipeEntity(
+      keyRange,
+      title,
+      imgURL,
+      description,
+      prepTime,
+      difficulty,
+      ingredients,
+      stepList,
+      request
+    );
+    Document recipeDocument = buildRecipeDocumentForIndexing(recipeEntity);
+
+    datastore.put(recipeEntity);
+    index.put(recipeDocument);
+  }
+
+  private Entity buildRecipeEntity(
+    KeyRange keyRange,
+    String title,
+    String imgURL,
+    String description,
+    int prepTime,
+    String difficulty,
+    ArrayList<String> ingredients,
+    ArrayList<String> stepList,
+    HttpServletRequest request
+  ) {
+    Entity recipeEntity = new Entity(keyRange.getStart());
+
+    recipeEntity.setProperty("title", title);
+    recipeEntity.setProperty("index_title", title.toLowerCase());
```

Comment:
Maybe the name of the index should be a class variable, because now we have different names for it.

Reply:
how do you mean?

Reply:
The name of the index can be set at the beginning of the class.